Share 114 implement docker

fladens opened the pull request:
Implementation of Docker

HCrane commented on docker/config/parameters.yml.dist:
```diff
@@ -0,0 +1,44 @@
+    database_name:     symfony
+    database_user:     root
+    database_password: root
+    database_server_v: 10.3.11
```
it should be `database_server_v: mariadb-10.3.11`

HCrane commented on docker/php/LICENSE:
```diff
@@ -0,0 +1,20 @@
+The MIT License (MIT)
```
Out of curiosity is this license needed?

HCrane commented:
Something i noticed when trying to do composer updates -> the `composer.json` and `composer.lock` do not get changed so a real update is not done only inside the containers...this should change.

fladens committed:
SHARE-114 Implement Docker

To have a more reliable way testing and developing on different systems.
And as preparation for parallel testing.

Added a docker folder with
- compose files and Dockerfile for test and dev environment
- app folder with scripts for running inside container
- config folder for parameters and behat setup
- README for guide how to use docker with catroweb

HCrane commented:
Rebased to latest develop, all test run fine natively so we are good to go for merge!
